File: app/src/main/java/com/gu/swiperefreshplus/PagerAdapter.java
package com.gu.swiperefreshplus;

import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;

import com.gu.swiperefreshplus.fragment.ListFragment;
import com.gu.swiperefreshplus.fragment.NormalViewFragment;
import com.gu.swiperefreshplus.fragment.RecycleFragment;

/**
 * Created by GUHY on 2017/4/5.
 */

public class PagerAdapter extends FragmentPagerAdapter {
    public PagerAdapter(FragmentManager fm) {
        super(fm);
    }

    @Override
    public Fragment getItem(int position) {
        Fragment fragment=null;
        switch (position){
            case 0:
                fragment= new RecycleFragment();
                break;
            case 1:
                fragment=new ListFragment();
                break;
            case 2:
                fragment=new NormalViewFragment();
                break;

        }
        return fragment;
    }

    @Override
    public int getCount() {
        return 3;
    }

    @Override
    public CharSequence getPageTitle(int position) {
        switch (position){
            case 0:
                return "RecycleView";
            case 1:
                return "ListView";
            case 2:
                return "NormalView";
        }
        return "";
    }
}

File: app/src/main/java/com/gu/swiperefreshplus/PagerAdapter.kt
package com.gu.swiperefreshplus

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter

import com.gu.swiperefreshplus.fragment.ListFragment
import com.gu.swiperefreshplus.fragment.NormalViewFragment
import com.gu.swiperefreshplus.fragment.RecycleFragment

/**
 * Created by GUHY on 2017/4/5.
 */

class PagerAdapter(fm: androidx.fragment.app.FragmentManager) : androidx.fragment.app.FragmentPagerAdapter(fm) {

    override fun getItem(position: Int): androidx.fragment.app.Fragment? {
        var fragment: androidx.fragment.app.Fragment? = null
        when (position) {
            0 -> fragment = RecycleFragment()
            1 -> fragment = ListFragment()
            2 -> fragment = NormalViewFragment()
        }
        return fragment
    }

    override fun getCount(): Int {
        return 3
    }

    override fun getPageTitle(position: Int): CharSequence {
        when (position) {
            0 -> return "RecycleView"
            1 -> return "ListView"
            2 -> return "NormalView"
            else-> return ""
        }
    }
}
